### PR TITLE
CP-41450: Moved C samples

### PR DESCRIPTION
**Pull request by kc284**
Please review only the 2nd commit, the first one `Add 'ocaml/sdk-gen/'...` is transfer from the xen-api repo.

### CODE REVIEW

File: c/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+on Debian. You may choose to use `curl` in your application, just as we have for
+these test programs, though it is not required to do so, and you may use a
+different network layer if you prefer.
+
+## How to compile
+
+Once you have installed libxenserver, run `make` in this folder.
+
+To build on Windows with [cygwin](https://www.cygwin.com) run `make CYGWIN=1`.
+(Remember that cygwin expects a libxenserver.dll, so ensure you have installed
```
Comment (kc284) [inline c/README.md:20]: Note that this statement will need a small change in the library's Makefile, I'll do it when I remove the samples from the xen-api repo.

File: c/Makefile [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#
+# Copyright (c) Citrix Systems, Inc.
```
Comment (katshann) [inline c/Makefile:2]: Does this need to change to "Cloud Software Group Holdings, Inc." now or is this for future when legal tell us what to call ourselves?

**Review by katshann — APPROVED** [review]

**Review by katshann — APPROVED** [review]

File: c/Makefile (hunk 2 above)
Comment (kc284) [reply c/Makefile:2]: You're right, the new copyright is to be used for new files and this Makefile is technically new while the other files are transfers from a different repo. I'll correct it.

Comment (kc284) [reply c/Makefile]: Done.

**Pull request merged**